From ca61129542c7cf2d86589251548eb41dd172f9ae Mon Sep 17 00:00:00 2001
From: Dave Stevenson <dave.stevenson@raspberrypi.com>
Date: Wed, 29 Apr 2020 22:05:09 +0100
Subject: [PATCH 236/634] media: bcm2835-unicam: Do not stop streaming in
 unicam_release

unicam_release calls _vb2_fop_release, which will call stop_streaming
if that particular node was streaming. Calling it unconditionally (as
the code was) means that if a second handle was opened eg to alter
a setting, on closing that connection it also stopped Unicam.

Signed-off-by: Dave Stevenson <dave.stevenson@raspberrypi.com>
---
 drivers/media/platform/bcm2835/bcm2835-unicam.c | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/drivers/media/platform/bcm2835/bcm2835-unicam.c b/drivers/media/platform/bcm2835/bcm2835-unicam.c
index 0e1d525a0b35..c7e211d04ed3 100644
--- a/drivers/media/platform/bcm2835/bcm2835-unicam.c
+++ b/drivers/media/platform/bcm2835/bcm2835-unicam.c
@@ -2204,9 +2204,6 @@ static int unicam_release(struct file *file)
 	if (fh_singular)
 		v4l2_subdev_call(sd, core, s_power, 0);
 
-	if (node->streaming)
-		unicam_stop_streaming(&node->buffer_queue);
-
 	node->open--;
 	mutex_unlock(&node->lock);
 
-- 
2.33.1

